packages/zowe-explorer-api/src/vscode/ui/utils/TableBuilder.ts

```diff
@@ -0,0 +1,224 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { ExtensionContext } from "vscode";
+import { Table } from "../TableView";
+import { TableMediator } from "./TableMediator";
+
+/**
+ * A builder class for quickly instantiating {@link Table.View} instances.
+ * Especially useful for building multiple tables with similar layouts or data.
+ *
+ * @remarks
+ *
+ * ## Building a table
+ *
+ * Developers can build a table using the helper methods provided by the builder.
+ * Once a table is ready to be built, use the {@link TableBuilder.build} function to create
+ * a new {@link Table.View} instance with the given configuration.
+ *
+ * ## Sharing tables
+ *
+ * Share a table during the build process by using the {@link TableBuilder.buildAndShare} function.
+ * This function will create the {@link Table.View} instance while also adding it to the mediator
+ * in Zowe Explorer's extender API. Extenders who would like to contribute to shared tables can do
+ * so by accessing the table by its unique ID.
+ */
+export class TableBuilder {
+    private context: ExtensionContext;
+    private data: Table.ViewOpts = {
```

Comment:
Wondering if we should just leave this as undefined here and call `this.reset()` as part of the constructor.

it's just to reduce duplication 😋 

I know it's possible to want to access the data of a tableBuilder without initializing it first, so maybe this is not that important to change 😋

Reply:
I believe I assigned `data` to be an "empty-like" structure so that by default, the TableBuilder renders an empty table with no contents or title. If we set `data` to `undefined`, a user can create a TableBuilder instance and then call `build` on it, but the resulting table would not render any contents in the editor or panel.

Reply:
How would one create a TableBuilder instance?
`new TableBuilder(context)` ?


If that's the case, and assuming we add a `this.reset()` call in the constructor...
```
    public constructor(context: ExtensionContext) {
        this.reset();
        this.context = context;
    }
```
 I think that `this.data` will be populated if even if the default value is not defined at the class level.
 
 ---
 
 Otherwise, where do we plan to use the `reset()` function ourselves?  (playing devil's advocate) 😈

Reply:
That's a good point, sorry I overlooked the aspect of adding a `this.reset` call - I'll adjust that so that it uses reset instead 😋 

Developers (such as ourselves or extenders) can also use the `reset` function to clean up a builder between creation of two or more tables. It could also be useful to keep around in-memory and leveraged for making similar tables, calling `reset` between each build.

Reply:
Thanks for the suggestion - addressed in 78e27ba